src/include/mpir_comm.h

```diff
@@ -195,6 +195,7 @@ struct MPIR_Comm {
 
     int revoked;                /* Flag to track whether the communicator
                                  * has been revoked */
+    int vci;                    /* A sequence number used in vci hashing */
```

Comment:
VCI is a `CH4` object. It shouldn't sit here.

Reply:
How about call it `seq`?

Reply:
This number is set in `MPIR_Comm_init` since we want a unique number for every user-level communicator; sub-comms (node_root_comm, local_comm) should inherit the same (sequence) number as parent. `MPID_Comm_create_hook` is being called for each comm including subcomm, so it is a bit awkward to manage.

How do you do it in your branch? Do you check whether a comm is a subcomm?

Reply:
I'll rename it to `seq` for now, and add `seq_list` for allgather. I'll keep in the MPIR-layer for now. We can easily move to ch4 in a separate PR.

Reply:
done (to-be-pushed).

Reply:
Could we move this piece down to the CH4 layer? I am not sure what its purpose is in the `MPIR` layer. It seems unnecessary to have another PR just to move it down to the CH4 layer when it needs to be done here.

Reply:
It is not clean to implement it in ch4 layer. The ch4 layer only can do this in `comm_creation_hook`, which is called for both MPI-layer comm and its sub-comms. We  want the seq number to be unique at MPI-layer and its sub-comms need to share the same seq number as its parent. Currently this is easily managed at `MPIR_Comm_init`.

Having a sequence number for MPI-comm objects in itself has nothing to do with devices. CH4 use it for vci, but it can be used for any other purpose as well.

Reply:
Is it not clean because we don't have a reference to the parent comm of a sub-comm in `MPID`?

Reply:
I don't think there is.